fdbserver/include/fdbserver/workloads/MetaclusterConsistency.actor.h

```diff
@@ -325,8 +335,19 @@ class MetaclusterConsistencyCheck {
 		}
 
 		auto& expectedTenantGroups = self->managementMetadata.clusterTenantGroupMap[clusterName];
-		ASSERT(dataClusterTenantGroups.results.size() == expectedTenantGroups.size());
-		for (auto [name, entry] : dataClusterTenantGroups.results) {
+		if (!self->allowPartialMetaclusterOperations) {
```

Comment:
This block and the creation of `groupExpectedTenantCounts` above is the meat of this change.